### PR TITLE
[SYCL][PI][L0] Uplift L0 Loader from 1.4.1 to 1.8.8

### PR DESCRIPTION
**Pull request by yanfeng3721**

### CODE REVIEW

**Review by againull — APPROVED** [review]

**Comment by yanfeng3721** [comment]
```
Failed Tests (1):
  SYCL :: Basic/barrier_order.cpp
```
Failed test is not related to current PR and due to https://github.com/intel/llvm-test-suite/issues/1371

**Comment by yanfeng3721** [comment]
@bader , could you please help to merge the PR? Thanks.

**Comment by bader** [comment]
@againull, is it okay to merge this PR? I see that you approved but haven't merged it.

File: sycl/plugins/level_zero/CMakeLists.txt [hunk 1]
```diff
@@ -24,7 +24,7 @@ if (NOT DEFINED LEVEL_ZERO_LIBRARY OR NOT DEFINED LEVEL_ZERO_INCLUDE_DIR)
   if (WIN32)
     set(LEVEL_ZERO_LOADER_SOURCE_VERSION v1.2.3)
   else()
-    set(LEVEL_ZERO_LOADER_SOURCE_VERSION v1.4.1)
+    set(LEVEL_ZERO_LOADER_SOURCE_VERSION v1.8.8)
   endif()
```
Comment (bader) [inline sycl/plugins/level_zero/CMakeLists.txt:28]: Why do we use different version on Windows and non-Windows platforms?

Comment (againull) [reply sycl/plugins/level_zero/CMakeLists.txt:28]: Yes, it is ok to merge this.

**Comment by againull** [comment]
Basic/barrier_order.cpp is disabled for HIP backend - https://github.com/intel/llvm-test-suite/pull/1376

**Comment by againull** [comment]
> @againull, is it okay to merge this PR? I see that you approved but haven't merged it.

https://github.com/intel/llvm-test-suite/pull/1376

**Comment by againull** [comment]
> @againull, is it okay to merge this PR? I see that you approved but haven't merged it.

Yes, it is ok to merge this.

File: sycl/plugins/level_zero/CMakeLists.txt (hunk 1 above)
Comment (smaslov-intel) [reply sycl/plugins/level_zero/CMakeLists.txt:28]: I don't know the answer to that question, but have another one: why did we need to change the Linux version but not the Windows version?

Comment (bader) [reply sycl/plugins/level_zero/CMakeLists.txt:28]: IIRC, there was some problem with using newer loader + Windows GPU driver combination. I wonder if this problem still exists, or we can switch back to using the same loader version for all platforms.
@yanfeng3721, could you test v1.8.8 on Windows, please?

**Pull request merged**